server/src/repositories/job.repository.ts

```diff
@@ -249,24 +250,42 @@ export class JobRepository implements IJobRepository {
   }
 
   private getJobOptions(item: JobItem): JobsOptions | null {
+    let opts = {};
     switch (item.name) {
+      case JobName.NOTIFY_ALBUM_UPDATE: {
+        opts = { jobId: item.data.id, delay: item.data?.delay };
+        break;
+      }
       case JobName.STORAGE_TEMPLATE_MIGRATION_SINGLE: {
-        return { jobId: item.data.id };
+        opts = { jobId: item.data.id };
+        break;
       }
       case JobName.GENERATE_PERSON_THUMBNAIL: {
-        return { priority: 1 };
+        opts = { priority: 1 };
+        break;
       }
       case JobName.QUEUE_FACIAL_RECOGNITION: {
-        return { jobId: JobName.QUEUE_FACIAL_RECOGNITION };
-      }
-
-      default: {
-        return null;
+        opts = { jobId: JobName.QUEUE_FACIAL_RECOGNITION };
+        break;
       }
     }
+    return opts;
   }
 
   private getQueue(queue: QueueName): Queue {
     return this.moduleReference.get<Queue>(getQueueToken(queue), { strict: false });
   }
+
+  public async removeJob(jobId: string, name: JobName): Promise<IEntityJob | undefined> {
+    const existingJob = await this.getQueue(JOBS_TO_QUEUE[name]).getJob(jobId);
+    if (existingJob === undefined) {
+      return undefined;
+    }
+    try {
+      await existingJob.remove();
+    } catch {
+      return undefined;
+    }
```

Comment:
This is sketchy. Best to check the error to make sure it's what you expect (which I assume is that the job doesn't exist)

Reply:
Yeah in this case I didn't want job doesn't exist errors to bubble up, which I think would be rare since we just got the job a few lines above, so I just returned `undefined` instead, are you saying I should catch the specific job not found exception, but let other errors bubble up?

Reply:
Yes, something like a connection error etc. shouldn't be ignored.

Reply:
I'm not sure how to handle this one, it look like bullmq internally only throws `Error`s, https://github.com/taskforcesh/bullmq/blob/9307fef3e5f9e5f929fd8755078913a33c2dc783/src/classes/job.ts#L601-L603 and https://github.com/taskforcesh/bullmq/blob/9307fef3e5f9e5f929fd8755078913a33c2dc783/src/classes/scripts.ts#L1423

Reply:
Oh, this is the fun part about JS. *Everyone* just throws an `Error` so you have to check the message etc. to know what kind of error it is.

Reply:
Would you check it by making a new error type or just checking if the error string contains `Missing key for job`?

Reply:
String matching is fine